pkg/xds/generator/admin_proxy_generator.go

```diff
@@ -41,13 +63,39 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 
 	resources := core_xds.NewResourceSet()
 
+	for _, se := range staticEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+	}
+
+	for _, se := range staticTlsEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+
+		token, err := ctx.EnvoyAdminClient.GenerateAPIToken(proxy.Dataplane)
+		if err != nil {
+			return nil, err
+		}
+		se.Header = "Authorization"
+		se.HeaderExactMatch = fmt.Sprintf("Bearer %s", token)
+	}
+
+	keyPair, err := NewSelfSignedCert("admin", tls.ServerCertType, "localhost")
+	if err != nil {
+		return nil, err
+	}
+
 	// We bind admin to 127.0.0.1 by default, creating another listener with same address and port will result in error.
 	if proxy.Dataplane.Spec.GetNetworking().Address != "127.0.0.1" {
```

Comment:
I think we need similar logic in admin client? If address is `127.0.0.1` that means that CP and DP is on the same machine and should not use https, because envoy admin is exposed directly

optional to do since admin client does not work on universal right now anyways

Reply:
I will skip this now :)